api/audit/models.py

```diff
@@ -79,9 +79,9 @@ def environment_document_updated(self) -> bool:
         return "send_environments_to_dynamodb" not in skip_signals_and_hooks
 
     @property
-    def history_record(self):
+    def history_record(self) -> typing.Optional[Model]:
         klass = self.get_history_record_model_class(self.history_record_class_path)
-        return klass.objects.get(id=self.history_record_id)
+        return klass.objects.filter(history_id=self.history_record_id).first()
```

Comment:
The only test that had to change is the one where the model is mocked. I'd surely prefer to have a more integrated test for this.

Reply:
I stand corrected, this code gets called by the new tests.

Reply:
As discussed offline, needs to change `first` to `get` to avoid returning `None`, though.

Reply:
I took a slightly different approach and handled the `None` return from this method in the calling method instead but this should be resolved now.